### ISSUE TITLE
Incorrect default background color in high-contrast-mode

### ISSUE THREAD

**Issue by nonplus** [dequelabs/axe-core#2267]
When [high-contrast mode is enabled on Microsoft Windows](https://support.microsoft.com/en-us/help/13862/windows-10-use-high-contrast-mode#:~:text=Turn%20high%20contrast%20mode%20on,colors%20on%20the%20screen%20change), browser background defaults to **black** rather than **white**.    Elements whose background stack doesn't have at least one `background-color` definition will calculate their contrast ratio against an [incorrect background color](https://github.com/dequelabs/axe-core/blob/7f4a75ae739345610d67a1d27d3ecf048887692d/lib/commons/color/get-background-color.js#L74).
```html
<html>
<body>
  <a href="http://example.com">Some Link</a>
</body>
</html>
```
```
ERROR: (color-contrast) Ensures the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds 

SUMMARY: Fix any of the following:
  Element has insufficient color contrast of 3.25 (foreground color: #8080ff, background color: #ffffff, ...)
```

Expectation: In dark high-contrast-mode, background color should default to black rather than white.

Actual: Background color always defaults to white.

Motivation: Current behavior is incorrect

<pre><code>
axe-core version: 3.5.4
</code></pre>

**Comment by WilcoFiers** [comment]
Axe-core is not designed to accurately test in an environment where custom styles are applied. Custom styles can create any number of accessibility issues that axe-core could pick up. Practically too, I don't think it is possible for axe-core to detect the background color in high contrast mode.

That said, axe-core does lack rules to check for missing foreground and background colors. If the text color is set, there should be a background, and vice versa. That has been on our list for a while, but there isn't an open issue for it. I'll make sure to do that.

**Comment by nonplus** [comment]
Detecting the default background color is fairly straightforward.  The following is based off [various search results](https://www.google.com/search?q=detect+high+contrast+mode+javascript):
```javascript
/**
 * Return the default document background color.
 *
 * Note: When high-contrast mode is enabled on Windows, the default color
 * may be something other than white.
 *
 * @return String containing hex or rgb color specification.
 */
function defaultBackgroundColor() {
  // Create element and assign it a custom color #123456
  const elt = document.createElement("a");
  elt.style.backgroundColor = "rgb(18,52,86)";

  // Get the computed background color
  document.documentElement.appendChild(elt);
  let strBackgroundColor = document.defaultView
    ? document.defaultView.getComputedStyle(elt, null).backgroundColor
    : elt.currentStyle.backgroundColor;
  // Normalize color representation
  strBackgroundColor = strBackgroundColor.toLowerCase().replace(/ /g, "");
  document.documentElement.removeChild(elt);

  // Check whether background color was overridden in HCM
  if (strBackgroundColor !== "#123456" && strBackgroundColor !== "rgb(18,52,86)") {
    // Return HCM background color
    return strBackgroundColor;
  }

  // Default to white
  return "#ffffff";
}
```

Let me know if you are interested in a PR.

**Comment by WilcoFiers** [comment]
@nonplus Are you still interested in submitting a PR? It's been a bit busy with the 4.0 release, but now that's behind us. This would be a neat addition. We'd definitely want to cache the result, other then that, this looks like it might work.

**Comment by straker** [comment]
The provided code may not work as expected. Trying this out in Windows HCM Black in Chrome I get back a default color of #ffffff. However, the same code in IE11 and Firefox does return the default color. Since Chrome has problems I'm not sure we could implement the feature as it would give different results based on the browser used.

**Comment by straker** [comment]
Due to the inconsistency in the browsers ability to correctly detect the color, we won't be able to handle this at the browser level.
